### DISCUSSION TITLE
Sensore con media di ultime due ore

### DISCUSSION THREAD

**Discussion by zanac** [indomus/forum#2534] [GENERALE - Discussioni generiche]
Ciao,

sto cercando il modo di avere un sensore (basato su history_stats?) che contenga la temperatura media di due ore precedenti saltando gli ultimi 10 minuti... è fattibile secondo voi? Sto smanettando da due ore con history_stats senza vernirne a capo!

**Comment by 50494554524F** [comment]
vedi se questo può servire altrimenti altro non c'è... dovresti andare direttamente di sql

https://www.home-assistant.io/integrations/statistics/

**Comment by zanac** [comment]
Ho trovato ha-average (https://github.com/Limych/ha-average)

Ecco il risultato, ho costruito una mappa termica che va dal blu (quando è inverno e fa freddo) al rosso... che mostra la settimana in corso in modo da avere a colpo d'occhio la sensazione di quanto fa caldo (o freddo).

![image](https://github.com/indomus/forum/assets/21194919/5da5d16e-18d0-4e2d-a468-2d2050380395)

**Reply by indomus** [reply]
C'era già questo: https://www.home-assistant.io/integrations/statistics/

**Reply by 50494554524F** [reply]
ottima segnalazione ha-average! fortuna che avevo pure cercato in hacs :D

(la mappa l'hai fatta te o è un altro componente?!)

**Reply by zanac** [reply]
La mappa ho descritto come è realizzata nel thread https://github.com/indomus/forum/discussions/2536
Faccio uso di un altro componente che permette di fare codice html in una card, si chiama html-template-card, in pratica però è come se l'avessi fatta tutta io dato che ho dovuto scrivere tutto il codice html da solo.

**Reply by 50494554524F** [reply]
mi era sfuggito, grazie... questo componente me lo ricordavo, bon

**Comment by zanac** [comment]
> C'era già questo: https://www.home-assistant.io/integrations/statistics/

Ho provato in tutti i modi ma non ne davo fuori... mi fai un esempio per ottenere un sensore virtuale che estragga il valore medio di, esempio, sensor.outsidetemp calcolato dalle 02:00 alle 03:59 di due giorni fa? Con questo componente uso questa sintassi...
  - platform: average                                                                                                                                  
    name: 'ext_temp_avg_02_02'                                                                                                                         
    start: '{{ now().replace(hour=2).replace(minute=0).replace(second=0) - timedelta(hours=48) }}'                                                     
    duration:                                                                                                                                          
      hours: 2                                                                                                                                         
    entities:                                                                                                                                          
      - sensor.ca350_outside_temperature
